### ISSUE TITLE
Node.js build requirement breaks building on 32-bit PowerPC

### ISSUE THREAD

**Issue by wyatt8740** [svaarala/duktape#2446]
node.js does not support 32-bit PowerPC on account of V8 not having a bytecode generator for it. At least on Linux.

There is also [no node.js package available in Debian Sid on 32-bit PPC](http://ftp.ports.debian.org/debian-ports/pool-powerpc/main/n/) as a result.

If you decide to test this, you can get an installer for 32-bit PPC from https://cdimage.debian.org/cdimage/ports/current/ - I believe it should run under QEMU if you (understandably) don't have appropriate PPC32 hardware. I am using a Powerbook G4 for testing.

The bottom line: the move from python to a node-based build system makes it nearly impossible to use on PowerPC, even though once built it will work fine.

**Comment by svaarala** [comment]
The short answer is that in the current state:
- You could try to use an existing release, and its default configuration in which case a configure step (and thus Python/Node.js) is not needed.
- Failing that, you should run the dist/configure step on an x64 host and then use the result source/header files in the target build. The compilation itself doesn't require Node.js, obviously. Unless your configuration changes all the time, this may be quite workable.

I switched from Python to Node.js for a few reasons:
1. Python2 -> Python3 transition is painful, and I don't want to come to that transition again at some future point in time.
2. Using Javascript is more natural since this is after all a Javascript project.
3. Node.js is faster than Python, so configure/compile cycles are a bit faster (nice, but not critical).
4. And finally, having an option to self host the configuration process on Duktape itself.

Self hosting would make Duktape configure process portable to almost all targets running Duktape itself. It's actually functional in practice (though not available readily in master), and I hope to preserve and polish the self hosting capability into 3.x release.

**Comment by Thaodan** [comment]
From my point of view nodejs isn't just an issue specific to unsupported platforms but in general.
Just to built duktape now another build dependency is needed which takes at least about an hour to build. We have to build a Javascript engine to built another..

That python is slower does not impact the build process enough to matter.

duktape self hosting would only work if  it can bootstrap itself through another language that is
likely already installed on the system and doesn't require hour(s) of building.

Javascript for building software doesn't sound like a good idea. Instead of rolling a build system reusing one such as CMake would be better IMHO.
Not just for building but also for development to integrate with other tooling such as language-servers.
